packages/components/src/components/tablist/tablist.component.ts

```diff
@@ -0,0 +1,430 @@
+ * @cssproperty --mdc-container-button-padding - Padding for when an arrow button is visible
+ */
+class TabList extends Component {
+  /**
+   * ID of the active tab, defaults to the first tab if not provided
+   */
+  @property({ type: String, attribute: 'active-tabid', reflect: true })
+  activeTabId?: string;
+
+  @query('.container')
+  private tabsContainer?: HTMLDivElement;
+
+  @queryAssignedElements({ selector: 'mdc-tab' })
+  private tabs?: Tab[];
+
+  /**
+   * Instead of using 'left' and 'right' for notation of the arrow buttons, we use 'forward' and 'backward'.
+   * Since the elements of the tablist gets flipped in RTL layouts, the arrow buttons also flip.
+   * For both RTL and LTR layouts, the forward arrow button moved the tabs in the forward direction.
+   * i.e. the forward arrow button moves the tabs to the right in LTR layouts and to the left in RTL layouts.
+   */
+
+  /**
+   * @internal
+   */
+  @state() private showForwardArrowButton: boolean = false;
+
+  /**
+   * @internal
+   */
+  @state() private showBackwardArrowButton: boolean = false;
+
+  private get isRtl(): boolean {
+    return (
+      document.querySelector('html')?.getAttribute('dir') === 'rtl' || window.getComputedStyle(this).direction === 'rtl'
+    );
+  }
+
+  constructor() {
+    super();
+
+    this.addEventListener('keydown', this.handleKeydown);
+    // Reason for assertion below: https://github.com/microsoft/TypeScript/issues/28357
+    this.addEventListener('activechange', this.handleNestedTabActiveChange as EventListener);
+  }
+
+  override connectedCallback() {
+    super.connectedCallback();
+    this.role = 'tablist';
+  }
+
+  /**
+   * Set the initial active tab after firstUpdated,
+   * since this.tabs (=slot) are only available after
+   * the first update
+   *
+   * Also set up the event listeners.
+   */
+  protected override firstUpdated(): void {
+    /**
+     * If there are no tabs, skip the initialization of the active tab and the event listeners.
+     * Then throw an error.
+     */
+    if (!this.tabs) { return; }
+    if (Array.isArray(this.tabs) && this.tabs.length === 0) {
+      throw new Error('The tablist component must have at least one child tab');
+    }
+
+    const tabIds = this.tabs.map((tab) => tab.tabId);
+    if (new Set(tabIds).size !== this.tabs.length) {
+      // eslint-disable-next-line no-console
+      console.error('The tabs inside the tab list must have unique tab ids');
+    }
+
+    const resizeObserver = new ResizeObserver((): void => {
+      this.handleArrowButtonVisibility();
+    });
+    resizeObserver.observe(this);
+
+    this.tabs?.forEach((tab) => {
+      tab.addEventListener('focus', this.handleFocus);
+      tab.addEventListener('mousedown', this.handleMousedown);
+    });
+
+    if (!this.activeTabId) {
+      this.activeTabId = getFirstTab(this.tabs)?.tabId;
+    }
+  }
+
+  /**
+   * Observe the tablist element for changes in the activetabid attribute.
```

Comment:
```suggestion
   * Observe the tablist element for changes in the activeTabId attribute.
```
Lets be consistant with the naming conventions.

Reply:
Updated